ChangeLog.md

```diff
@@ -1,5 +1,290 @@
+- Sort out some compatibility issues
+([#2511](https://github.com/haskell/haskell-language-server/pull/2511)) by @alanz
+- Fix ci cache for windows
+([#2507](https://github.com/haskell/haskell-language-server/pull/2507)) by @jneira
+- CI: caching: add early termination & run check on schedule
+([#2506](https://github.com/haskell/haskell-language-server/pull/2506)) by @Anton-Latukha
+- Fix tracing of recordDirtyKeys
+([#2505](https://github.com/haskell/haskell-language-server/pull/2505)) by @pepeiborra
+- Unhandled exceptions fixed
+([#2504](https://github.com/haskell/haskell-language-server/pull/2504)) by @pepeiborra
+- Build with GHC 9.2
+([#2503](https://github.com/haskell/haskell-language-server/pull/2503)) by @pepeiborra
+- Ignore stack.yamls in test cabal workflow
+([#2502](https://github.com/haskell/haskell-language-server/pull/2502)) by @jneira
+- small stack yaml updates to ease maintenance
+([#2501](https://github.com/haskell/haskell-language-server/pull/2501)) by @simonmichael
+- Automatically read in the doc version from the cabal file
+([#2500](https://github.com/haskell/haskell-language-server/pull/2500)) by @michaelpj
+- Disable alternate numbers format plugin temporary
+([#2498](https://github.com/haskell/haskell-language-server/pull/2498)) by @jneira
+- Revert "Send unhandled exceptions to the user (#2484)"
+([#2497](https://github.com/haskell/haskell-language-server/pull/2497)) by @jneira
+- Upgrade to new version of lsp libraries
+([#2494](https://github.com/haskell/haskell-language-server/pull/2494)) by @michaelpj
+- Fail if main or pre jobs are cancelled
+([#2493](https://github.com/haskell/haskell-language-server/pull/2493)) by @jneira
+- stack-9.0.1: update/cleanup
+([#2489](https://github.com/haskell/haskell-language-server/pull/2489)) by @simonmichael
+- Correctly handle LSP shutdown/exit
+([#2486](https://github.com/haskell/haskell-language-server/pull/2486)) by @pepeiborra
+- Fix hls-graph ide build with embed-files
+([#2485](https://github.com/haskell/haskell-language-server/pull/2485)) by @pepeiborra
+- Send unhandled exceptions to the user
+([#2484](https://github.com/haskell/haskell-language-server/pull/2484)) by @pepeiborra
+- Fix redundant import actions for names starting with _
+([#2483](https://github.com/haskell/haskell-language-server/pull/2483)) by @Ailrun
+- Update flake to use fourmolu plugin in GHC 9
+([#2482](https://github.com/haskell/haskell-language-server/pull/2482)) by @Ailrun
+- Delete some dead or deprecated settings
+([#2481](https://github.com/haskell/haskell-language-server/pull/2481)) by @michaelpj
+- Class plugin bump up
+([#2475](https://github.com/haskell/haskell-language-server/pull/2475)) by @Ailrun
+- Fix some pragma completion cases
+([#2474](https://github.com/haskell/haskell-language-server/pull/2474)) by @Ailrun
+- Minor org to contribution doc
+([#2472](https://github.com/haskell/haskell-language-server/pull/2472)) by @Anton-Latukha
+- Warn if TH and Mac and static binary
```

Comment:
Since we have rolled back the Musl change, we should tweak this warning to fire not only in Mac but also in Linux

Reply:
agree, will do in this pr

Reply:
done with 539f00, i've relaxed the warning as th evaluation works in some cases for linux